src/engine/tests/Test_MILPEncoder.h

```diff
@@ -1241,6 +1241,190 @@ class MILPEncoderTestSuite : public CxxTest::TestSuite
+        tableau.setDimensions( 1, 1 );
+        inputQuery.setLowerBound( 0, 1.2 );
+        tableau.setLowerBound( 0, 1.3 );
+
+        List<Tightening> tightenings;
+        round->getEntailedTightenings( tightenings );
+
+        for ( const auto &t : tightenings )
+        {
+            if ( t._type == Tightening::LB )
+            {
+                inputQuery.setLowerBound( t._variable, t._value );
+                tableau.setLowerBound( t._variable, t._value );
+            }
+            if ( t._type == Tightening::UB )
+            {
+                inputQuery.setUpperBound( t._variable, t._value );
+                tableau.setUpperBound( t._variable, t._value );
+            }
+        }
+        round->registerTableau( &tableau );
+        inputQuery.addNonlinearConstraint( round );
+
+        MILPEncoder milp( tableau );
+        TS_ASSERT_THROWS_NOTHING( milp.encodeInputQuery( gurobi, inputQuery ) );
+
+        TS_ASSERT_THROWS_NOTHING( gurobi.solve() );
+
+        TS_ASSERT( gurobi.haveFeasibleSolution() );
+
+        Map<String, double> solution;
+        double costValue;
+        TS_ASSERT_THROWS_NOTHING( gurobi.extractSolution( solution, costValue ) );
+        TS_ASSERT( FloatUtils::areEqual( solution["x1"], 1 ) );
+#else
+        TS_ASSERT( true );
+#endif // ENABLE_GUROBI
+    }
+
+    void test_encode_round_constraint2()
+    {
+#ifdef ENABLE_GUROBI
+
+        /*
+          1.5 <= x0 <= 2.49999
+          x1 = Round( x0 )
+        */
+        GurobiWrapper gurobi;
+
+        InputQuery inputQuery = InputQuery();
+        inputQuery.setNumberOfVariables( 2 );
+
+        RoundConstraint *round = new RoundConstraint( 0, 1 );
+
+        round->notifyLowerBound( 0, 1.5 );
+        round->notifyUpperBound( 0, 2.49999 );
+        MockTableau tableau = MockTableau();
+        tableau.setDimensions( 1, 1 );
+
+        List<Tightening> tightenings;
+        round->getEntailedTightenings( tightenings );
+
+        for ( const auto &t : tightenings )
+        {
+            if ( t._type == Tightening::LB )
+            {
+                inputQuery.setLowerBound( t._variable, t._value );
+                tableau.setLowerBound( t._variable, t._value );
+            }
+            if ( t._type == Tightening::UB )
+            {
+                inputQuery.setUpperBound( t._variable, t._value );
+                tableau.setUpperBound( t._variable, t._value );
+            }
+        }
+        round->registerTableau( &tableau );
+        inputQuery.addNonlinearConstraint( round );
+
+        MILPEncoder milp( tableau );
+        TS_ASSERT_THROWS_NOTHING( milp.encodeInputQuery( gurobi, inputQuery ) );
+
+        TS_ASSERT_THROWS_NOTHING( gurobi.solve() );
+
+        TS_ASSERT( gurobi.haveFeasibleSolution() );
+
+        Map<String, double> solution;
+        double costValue;
+        TS_ASSERT_THROWS_NOTHING( gurobi.extractSolution( solution, costValue ) );
+        TS_ASSERT( FloatUtils::areEqual( solution["x1"], 2 ) );
```

Comment:
Don't you need the case of equidistant from two integers, e.g. 1.5 <= x0 <= 2.5?

Reply:
Added test for this case.

Reply:
I checked the added the test "test_encode_round_constraint4" and the assertion below.

```
        TS_ASSERT( FloatUtils::areEqual( solution["x1"], 1 ) ||
                   FloatUtils::areEqual( solution["x1"], 2 ) ||
                   FloatUtils::areEqual( solution["x1"], 3 ) );
```

I thought this expectation of the first line meant x1 was 1.0 when x0 = 1.
If it's correct, you might have to add `FloatUtils::areEqual( solution["x1"], 1) ||` as well in test_encode_round_constraint1?

Reply:
In this test, we have x0 between 1.5 and 2.5. But since our encoding does **not** precisely encode the equidistant case, we should expect the MILP solver to return anything between 1, 2, and 3. 

There is no such ambiguity in test_encode_round_constraint1 though, because x0 is between 1.2 and 1.3, and x1 can only be equal to 1.

Reply:
The reason we can distinguish all cases but equidistant case is because in MILP we encode f = Round(b) as the following:     

f - b <= 0.5
b - f <= 0.5

where f is an integer.